src/main/java/net/neoforged/neoforge/transfer/handlers/wrappers/SimpleResourceHandler.java

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.transfer.handlers.wrappers;
+
+import java.util.Objects;
+import net.neoforged.neoforge.transfer.TransferAction;
+import net.neoforged.neoforge.transfer.handlers.resources.IResourceHandler;
+import net.neoforged.neoforge.transfer.handlers.templates.resource.ResourceStorageComponentHandler;
+import net.neoforged.neoforge.transfer.resources.IResource;
+import net.neoforged.neoforge.transfer.transaction.Transaction;
+import net.neoforged.neoforge.transfer.transaction.TransactionContext;
+import net.neoforged.neoforge.transfer.transaction.TransactionManager;
+
+/**
+ * A simple resource handler that wraps an {@link IResourceHandler} and provides a simplified interface without
+ * needing to manually start or stop a transaction.
+ * <p>
+ * <p>
+ * This is intended for use in simple cases where you do not need the full power of the transaction system. It is important to remember, that this is not an IResourceHandler itself.
+ */
+public final class SimpleResourceHandler<T extends IResource> {
```

Comment:
Just throwing a comment here so that we make sure towards the end of the review cycle to go through this class and make sure that all the javadocs match those of the methods they are delegating to. As currently they are missing things such as the throws declarations etc